File: web/src/entities/wallOfDeathStarter.ts
import { ImmutableVec2 } from '../math.js';
import { Entity } from '../entity.js';

export class WallOfDeathStarter extends Entity {
  private used = false;

  constructor() {
    super();
    this.setSize(new ImmutableVec2(10, 10 * 100));
  }

  getLayer(): number { return 1; }

  override onRespawn(): void { this.used = false; }

  override draw(_ctx: CanvasRenderingContext2D, _ox: number, _oy: number, _layer: number): void {}

  override update(): void {
    if (!this.used && this.myRoom.getHero().Collides(this)) {
      this.myRoom.broadcastStartWallOfDeath();
      this.used = true;
    }
  }
}

File: wii/src/entities/wall_of_death_starter.cpp
#include "entities/wall_of_death_starter.h"
#include "entities/hero.h"
#include "room.h"

WallOfDeathStarter::WallOfDeathStarter() {
    setSize(Vec2(10, 10 * 100));
}

void WallOfDeathStarter::onRespawn() {
    used_ = false;
}

void WallOfDeathStarter::draw(SDL_Renderer* /*renderer*/, int /*offsetX*/, int /*offsetY*/, int /*layer*/) {
    // Invisible entity
}

void WallOfDeathStarter::update() {
    if (!used_ && room_->getHero()->Collides(*this)) {
        room_->broadcastStartWallOfDeath();
        used_ = true;
    }
}
